src/middlewared/middlewared/plugins/failover_/event.py

```diff
@@ -1037,7 +1037,7 @@ def stop_apps(self):
 
         logger.info('Trying to gracefully stop docker service')
         try:
-            self.run_call('service.stop', 'docker')
+            self.run_call('service.stop', 'docker', job=True)
```

Comment:
This is being done in a thread, i don't think we need that anymore considering this is a job now. We can remove the thread and just trigger a job and not wait for it and it will try/attempt to gracefully shut it down in the background.

Reply:
In other words, do you want to not wait for the job to complete (i.e. `stop_apps` will not wait for docker to actually stop)? This will change the logic.

Reply:
```
        # We will try to give some time to docker to gracefully stop before zpools will be forcefully
        # exported. This is to avoid any potential data corruption.
        stop_docker_thread = threading.Thread(
            target=self.stop_apps,
            name='failover_stop_docker',
        )
```
It is a thread right now, and we do not wait for it to complete right now either. I was thinking that we can just trigger the job and call it a day now and no need to create a thread for this (obviously only create stop job if docker is configured)

Reply:
The advantage of the thread is that it waits for the job to complete and logs the docker stop failure message.